src/ring_theory/int/basic.lean

```diff
@@ -144,6 +144,21 @@ lemma nat_abs_lcm (i j : ℤ) : nat_abs (gcd_monoid.lcm i j) = int.lcm i j := rf
 
 end gcd_monoid
 
+lemma sqr_of_coprime {a b c : ℤ} (hc : c ≠ 0) (h : int.gcd a b = 1) (heq : a * b = c ^ 2) :
+  ∃ (a0 : ℤ), a0 ≠ 0 ∧ (a = a0 ^ 2 ∨ a = - (a0 ^ 2)) :=
```

Comment:
It seems strange to include the `≠ 0` conditions in the hypotheses and in the conclusion.

Reply:
It is very convenient for my particular application, but you are right, not so nice... I'll fix.